src/hostapi/wasapi/pa_win_wasapi.c

```diff
@@ -4787,23 +4833,43 @@ static PaError ReadStream( PaStream* s, void *_buffer, unsigned long frames )
             continue;
         }
 
+        // Get buffer of the Mono to Stereo mixer, otherwise use WASAPI device buffer directly
+        if (isMonoStereoConverter)
+        {
+            // Expand buffer if necessary (normally buffer is preallocated by ActivateAudioClient())
```

Comment:
I get that it is normally preallocated, but this comment should also explain why it is sometimes necessary to expand the buffer here. Ideally you should never be calling an allocator inside ReadStream.

Reply:
I updated comments in places where `ReallocateMonoMixerBuffer` is called but prefer to leave those calls there because WASAPI is very driver dependent and they are more for a fail-safe operation.